lib/streamlit/type_util.py

```diff
@@ -227,6 +232,18 @@ def get_fqn_type(obj: object) -> str:
     _NUMPY_ARRAY_TYPE_STR,
 )
 
+# We show a special "UnevaluatedDataFrame" warning for cached funcs
+# that attempt to return one of these unserializable types:
+UNEVALUATED_DATAFRAME_TYPES = (
+    _SNOWPARK_TABLE_TYPE_STR,
```

Comment:
see this comment about sorting [here](https://github.com/streamlit/streamlit/pull/8506/files#r1580871049)